src/metrics_server/feature_metrics.ts

```diff
@@ -66,10 +66,22 @@ export function isValidFeatureMetricsReport(obj: any): obj is DailyFeatureMetric
     return false;
   }
 
-  // Validate individual feature records.
+  // Validate the server data limit feature
   if (typeof obj.dataLimit.enabled !== 'boolean') {
     return false;
   }
 
+  // Validate the per-key data limit feature
+  const perKeyType = typeof obj.dataLimit.perKeyLimitCount;
+  if (perKeyType !== 'undefined') {
```

Comment:
nit: may be clearer to check `obj.dataLimit.perKeyLimitCount === undefined`.

Reply:
Done.  I restructured the branching logic as well and I think it's a whole lot clearer now, so thanks for the prod!